src/core/ext/filters/client_channel/lb_policy/grpclb/load_balancer_api.h

```diff
@@ -33,69 +35,40 @@
 
 namespace grpc_core {
 
-typedef grpc_lb_v1_LoadBalanceRequest grpc_grpclb_request;
-typedef grpc_lb_v1_LoadBalanceResponse grpc_grpclb_response;
-typedef grpc_lb_v1_InitialLoadBalanceResponse grpc_grpclb_initial_response;
-typedef google_protobuf_Duration grpc_grpclb_duration;
-typedef google_protobuf_Timestamp grpc_grpclb_timestamp;
-
-typedef struct {
-  int32_t size;
-  char data[GRPC_GRPCLB_SERVER_IP_ADDRESS_MAX_SIZE];
-} grpc_grpclb_server_ip_address;
-
 // Contains server information. When the drop field is not true, use the other
 // fields.
-typedef struct {
-  grpc_grpclb_server_ip_address ip_address;
+struct GrpcLbServer {
+  int32_t ip_size;
+  char ip_addr[GRPC_GRPCLB_SERVER_IP_ADDRESS_MAX_SIZE];
   int32_t port;
   char load_balance_token[GRPC_GRPCLB_SERVER_LOAD_BALANCE_TOKEN_MAX_SIZE];
   bool drop;
-} grpc_grpclb_server;
-
-typedef struct {
-  grpc_grpclb_server** servers;
-  size_t num_servers;
-} grpc_grpclb_serverlist;
-
-/**
- * Create a request for a gRPC LB service under \a lb_service_name.
- * \a lb_service_name should be alive when returned request is being used.
- */
-grpc_grpclb_request* grpc_grpclb_request_create(const char* lb_service_name,
-                                                upb_arena* arena);
-grpc_grpclb_request* grpc_grpclb_load_report_request_create(
-    grpc_core::GrpcLbClientStats* client_stats, upb_arena* arena);
-
-/** Protocol Buffers v3-encode \a request */
-grpc_slice grpc_grpclb_request_encode(const grpc_grpclb_request* request,
-                                      upb_arena* arena);
-
-/** Parse (ie, decode) the bytes in \a encoded_grpc_grpclb_response as a \a
- * grpc_grpclb_initial_response */
-const grpc_grpclb_initial_response* grpc_grpclb_initial_response_parse(
-    const grpc_slice& encoded_grpc_grpclb_response, upb_arena* arena);
-
-/** Parse the list of servers from an encoded \a grpc_grpclb_response */
-grpc_grpclb_serverlist* grpc_grpclb_response_parse_serverlist(
-    const grpc_slice& encoded_grpc_grpclb_response);
-
-/** Return a copy of \a sl. The caller is responsible for calling \a
- * grpc_grpclb_destroy_serverlist on the returned copy. */
-grpc_grpclb_serverlist* grpc_grpclb_serverlist_copy(
-    const grpc_grpclb_serverlist* sl);
-
-bool grpc_grpclb_serverlist_equals(const grpc_grpclb_serverlist* lhs,
-                                   const grpc_grpclb_serverlist* rhs);
-
-bool grpc_grpclb_server_equals(const grpc_grpclb_server* lhs,
-                               const grpc_grpclb_server* rhs);
-
-/** Destroy \a serverlist */
-void grpc_grpclb_destroy_serverlist(grpc_grpclb_serverlist* serverlist);
 
-grpc_millis grpc_grpclb_duration_to_millis(
-    const grpc_grpclb_duration* duration_pb);
+  bool operator==(const GrpcLbServer& other) const {
```

Comment:
I <i>think</i> this equality check is UB unless we have a guarantee that every `GrpcLbServer` will be initialized with a memset to all zeros (due to possible garbage in possible padding fields). I think we need to either have such initialization, or do member by member comparison

Reply:
Done.